tilings/strategies/batch.py

```diff
@@ -0,0 +1,163 @@
+
+
+def all_factor_insertions(tiling: Tiling, ignore_parent: bool = False,
+                          **kwargs) -> Iterable[Rule]:
+    """
+    Insert all proper factor of the requirement or obstructions on the tiling.
+    """
+    yield from FactorInsertion(tiling).rules(ignore_parent)
+
+
+def requirement_corroboration(tiling: Tiling, ignore_parent: bool = True,
+                              **kwargs) -> Iterable[Rule]:
+    """
+    The requirement corroboration strategy.
+
+    The requirement corroboration strategy is a batch strategy that considers
+    each requirement of each requirement list. For each of these requirements,
+    the strategy returns two tilings; one where the requirement has been turned
+    into an obstruction and another where the requirement has been singled out
+    and a new requirement list added with only the requirement. This new
+    requirement list contains only the singled out requirement.
+
+    This implements the notion of partitioning the set of gridded permutations
+    into those that satisfy this requirement and those that avoid it. Those
+    that avoid the requirement, must therefore satisfy another requirement from
+    the same list and hence the requirement list must be of length at least 2.
+    """
+    yield from RequirementCorroboration(tiling).rules(ignore_parent)
+
+
+# -------------------------------------
+#   Row and column placement          |
+# -------------------------------------
+def row_placements(tiling, **kwargs):
+    yield from RequirementPlacement(tiling).all_row_placement_rules()
+
+
+def col_placements(tiling, **kwargs):
+    yield from RequirementPlacement(tiling).all_col_placement_rules()
+
+
+def row_and_col_placements(tiling, **kwargs):
+    req_placements = RequirementPlacement(tiling)
+    yield from req_placements.all_row_placement_rules()
+    yield from req_placements.all_col_placement_rules()
+
+
+def partial_row_placements(tiling, **kwargs):
+    req_placements = (RequirementPlacement(tiling, own_row=False),
+                      RequirementPlacement(tiling, own_col=False))
+    for req_placement in req_placements:
+        yield from req_placement.all_row_placement_rules()
+
+
+def partial_col_placements(tiling, **kwargs):
+    req_placements = (RequirementPlacement(tiling, own_row=False),
+                      RequirementPlacement(tiling, own_col=False))
+    for req_placement in req_placements:
+        yield from req_placement.all_col_placement_rules()
+
+
+def partial_row_and_col_placements(tiling, **kwargs):
+    req_placements = (RequirementPlacement(tiling, own_row=False),
+                      RequirementPlacement(tiling, own_col=False))
+    for req_placement in req_placements:
+        yield from req_placement.all_row_placement_rules()
+        yield from req_placement.all_col_placement_rules()
+
+
+# TODO: Clarify what is this function doing
+def requirement_list_placement(tiling, **kwargs):
```

Comment:
I do not really understand what this strategy is doing? Can we clarify the docstring?

Reply:
We decided to remove it